examples_tests/50.MeshPacking/main.cpp

```diff
@@ -1,190 +1,388 @@
-
-    // register the zip
-    device->getFileSystem()->addFileArchive("../../media/sponza.zip");
-
-    asset::IAssetLoader::SAssetLoadParams lp;
-    auto meshes_bundle = am->getAsset("sponza.obj", lp);
-    assert(!meshes_bundle.isEmpty());
-    auto mesh = meshes_bundle.getContents().begin()[0];
-    auto mesh_raw = static_cast<asset::ICPUMesh*>(mesh.get());
-
-    //saving cache to file
-    qnc->saveCacheToFile(asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10, fs, "../../tmp/normalCache101010.sse");
-
-    core::vector<ICPUMeshBuffer*> meshBuffers;
-    for (uint32_t i = 0u; i < mesh_raw->getMeshBufferCount(); i++)
-        meshBuffers.push_back(mesh_raw->getMeshBuffer(i));
-
-    DrawIndexedIndirectInput mdiCallParams;
-    SVertexInputParams vtxInputParams;
-    packMeshBuffers(driver, meshBuffers, vtxInputParams, mdiCallParams);
-
-    core::smart_refctd_ptr<IGPURenderpassIndependentPipeline> gpuPipeline;
-    setPipeline(driver, am, vtxInputParams, gpuPipeline);
-
-    //! we want to move around the scene and view it from different angles
-    scene::ICameraSceneNode* camera = smgr->addCameraSceneNodeFPS(0, 100.0f, 0.5f);
-
-    camera->setPosition(core::vector3df(-4, 0, 0));
-    camera->setTarget(core::vector3df(0, 0, 0));
-    camera->setNearValue(1.f);
-    camera->setFarValue(5000.0f);
-
-    smgr->setActiveCamera(camera);
-
-    uint64_t lastFPSTime = 0;
-    while (device->run() && receiver.keepOpen())
-    {
-        driver->bindGraphicsPipeline(gpuPipeline.get());
-
-        driver->beginScene(true, true, video::SColor(255, 0, 0, 255));
-
-        //! This animates (moves) the camera and sets the transforms
-        camera->OnAnimate(std::chrono::duration_cast<std::chrono::milliseconds>(device->getTimer()->getTime()).count());
-        camera->render();
-
-        driver->pushConstants(gpuPipeline->getLayout(), asset::ISpecializedShader::ESS_VERTEX, 0u, sizeof(core::matrix4SIMD), camera->getConcatenatedMatrix().pointer());
-        driver->drawIndexedIndirect(mdiCallParams.vtxBindings, mdiCallParams.mode, mdiCallParams.indexType, mdiCallParams.indexBuff.get(), mdiCallParams.indirectDrawBuff.get(), mdiCallParams.offset, mdiCallParams.maxCount, mdiCallParams.stride);
-
-        driver->endScene();
-    }
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#define _NBL_STATIC_LIB_
+#include <iostream>
+#include <cstdio>
+#include <nabla.h>
+
+//! I advise to check out this file, its a basic input handler
+#include "../common/QToQuitEventReceiver.h"
+#include "nbl/asset/CCPUMeshPackerV1.h"
+#include "nbl/asset/CCPUMeshPackerV2.h"
+
+using namespace nbl;
+using namespace core;
+using namespace asset;
+using namespace video;
+
+constexpr const char* SHADER_OVERRIDES[2] =
+{
+R"(
+#define _NBL_VERT_INPUTS_DEFINED_
+#define _NBL_VERT_MAIN_DEFINED_
```

Comment:
would be more clear if it were in the string with main(), than insert it into shader source just before `#ifndef _NBL_VERT_MAIN_DEFINED_`